specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,325 @@
+    // |Beta|Flighting with inner rings, automated testing|~4 weeks|
+    // |Dev|Automated testing, selfhosting to test new APIs and features|Weekly|
+    // |Canary|Automated testing, selfhosting to test new APIs and features|Daily|
+    [Flags] enum CoreWebView2RuntimeChannel
+    {
+        // The stable WebView2 Runtime that is released every 4 weeks.
+        Stable = 1,
+        // The Beta Runtime channel that is released every 4 weeks, a week before the stable release.
+        Beta = 2,
+        // The Dev Runtime channel that is released weekly.
+        Dev = 4,
+        // The Canary Runtime channel that is released daily.
+        Canary = 8,
+    };
+
+    runtimeclass CoreWebView2EnvironmentOptions
+    {
+        // ...
+
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2EnvironmentOptions6")]
+        {
+            // Sets the `RuntimeChannelPreferenceSet`, which is a mask of one or more
+            // `CoreWebView2RuntimeChannel`s. OR operation(s) can be applied to multiple
+            // `CoreWebView2RuntimeChannel`s to create a mask. . The default value is a
+            // a mask of all the channels. By default, the WebView2 loader searches for
+            // channels from most to least stable, using the first channel found on the
+            // device. When a `RuntimeChannelPreferenceSet` is provided, the WebView2
+            // loader will only search for the channels specified in the set.
+            // Set `ReverseChannelSearchOrder` to `true` to reverse the search order so
+            // that the loader searches for least stable build first. See
+            // `CoreWebView2RuntimeChannel` for descriptions of each channel. Environment creation
+            // fails if the loader is unable to find any channel from the
+            // `RuntimeChannelPreferenceSet` installed on the device. Use
+            // `CoreWebView2Environment.GetAvailableBrowserVersionStringWithOptions` to verify which channel
+            // is used.
+            //
+            // Examples:
+            // |   RuntimeChannelPreferenceSet   |   Default Channel Search Order   |   Reverse Channel Search Order   |
+            // | --- | --- | --- |
+            // |CoreWebView2RuntimeChannel.Beta \| CoreWebView2RuntimeChannel.Stable| WebView2 Runtime -> Beta | Beta -> WebView2 Runtime|
+            // |CoreWebView2RuntimeChannel.Canary \| CoreWebView2RuntimeChannel.Dev \| CoreWebView2RuntimeChannel.Beta \| CoreWebView2RuntimeChannel.Stable | WebView2 Runtime -> Beta -> Dev -> Canary | Canary -> Dev -> Beta -> WebView2 Runtime |
+            // |CoreWebView2RuntimeChannel.Canary| Canary | Canary |
+            // |CoreWebView2RuntimeChannel.Beta \| CoreWebView2RuntimeChannel.Canary \| CoreWebView2RuntimeChannel.Stable | WebView2 Runtime -> Beta -> Canary | Canary -> Beta -> WebView2 Runtime |
+            //
+            // If both a `BrowserExecutableFolder` and a `RuntimeChannelPreferenceSet` are
```

Comment:
Why have two properties (a flags mask, and a bool or enum for sort order), instead of one method for specifying the preferred channels in order?

Is it a common case to reverse the order from most/least stable, but keep the set of channels, so we're optimizing for that?

Reply:
In an earlier design the channel search order was automatically reversed when the set of channels was provided. We thought this design could be a little clearer and give the developer more control

Reply:
This API is easier for the scenarios we know and care about. We don't need a full list.